src/polyfill.rs

```diff
@@ -146,3 +146,18 @@ macro_rules! slice_as_array_ref_mut {
         }
     }
 }
+
+/// Returns a slice of arrays starting at the given pointer.
+macro_rules! ptr_as_array_slice {
+    ($ptr:expr, $block_len:expr, $block_num:expr) => {
```

Comment:
Sorry to take so long to get back to you.

The purpose of the `array_ref` stuff is to basically work around the lack of generics over integer values. I don't expect the Rust lang/lib teams to ever implement `slice_as_array_ref_chunks` but I do expect people can agree that it is useful to have some kind of functionality like that.

Anyway, one goal of the polyfill module to to expose only an interface that is "safe" (doesn't require `unsafe`), so we should try to find some solution that achieves that goal. I am definitely open to alternatives to what I suggested above.